iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttDirectMethod.java

```diff
@@ -27,14 +27,14 @@ class MqttDirectMethod extends Mqtt
     private static final String BACKSLASH = "/";
     private static final String QUESTION = "?";
 
-    private static final String METHOD = "$iothub/methods/";
-    private static final String POST = METHOD + "POST";
-    private static final String RES = METHOD + "res";
-    private static final String REQ_ID = QUESTION + "$rid=";
+    private static final String METHOD = "$iothub/methods";
+    private static final String POST = "POST";
+    private static final String RES = "res";
+    private static final String REQ_ID = "$rid=";
 
-    //Placement for $iothub/methods/POST/{method name}/?$rid={request id}
-    private static final int METHOD_TOKEN = 3;
-    private static final int REQID_TOKEN = 4;
+    //Placement for $iothub/methods/{clientId}/POST/{methodName}/?$rid={request id}
+    private static final int METHOD_TOKEN = 4;
+    private static final int REQID_TOKEN = 5;
```

Comment:
Now we have a new field "/{clientId}/" inserted between "$iothub/methods" and "POST", so move these 2 parts backward one place.